internal/rukpak/util/util.go

```diff
@@ -9,3 +27,89 @@ func MergeMaps(maps ...map[string]string) map[string]string {
 	}
 	return out
 }
+
+// PodNamespace checks whether the controller is running in a Pod vs.
+// being run locally by inspecting the namespace file that gets mounted
+// automatically for Pods at runtime. If that file doesn't exist, then
+// return DefaultSystemNamespace.
+func PodNamespace() string {
+	namespace, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return DefaultSystemNamespace
+	}
+	return string(namespace)
+}
+
+// FSToTarGZ writes the filesystem represented by fsys to w as a gzipped tar archive.
+// This function unsets user and group information in the tar archive so that readers
+// of archives produced by this function do not need to account for differences in
+// permissions between source and destination filesystems.
+func FSToTarGZ(w io.Writer, fsys fs.FS) error {
```

Comment:
This was originally in `tar.go`, did we want to keep it there, to keep the code _somewhat_ consistent with upstream rukpak?